clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/config/KubernetesConfiguration.java

```diff
@@ -59,19 +66,47 @@ public KubernetesProvider kubernetesProvider() {
   }
 
   @Bean
-  public KubernetesProviderSynchronizable kubernetesProviderSynchronizable(
-      KubernetesProvider kubernetesProvider,
-      AccountCredentialsRepository accountCredentialsRepository,
-      KubernetesCachingAgentDispatcher kubernetesCachingAgentDispatcher,
-      KubernetesConfigurationProperties kubernetesConfigurationProperties,
+  @ConditionalOnMissingBean(
+      value = KubernetesNamedAccountCredentials.class,
+      parameterizedContainer = AbstractCredentialsLoader.class)
+  public AbstractCredentialsLoader<KubernetesNamedAccountCredentials> kubernetesCredentialsLoader(
+      @Nullable
```

Comment:
Does Spring read this `@Nullable` annotation and inject `null` if there's no bean?  I've always used `Optional<MyClass>` which I know works in Spring, didn't realize that `@Nullable` worked as well.

Reply:
It does! IntelliJ started yelling at me and I gave in. I changed recently and there may still be some `Optional` laying around.

![Screen Shot 2020-09-25 at 9 50 35 AM](https://user-images.githubusercontent.com/352411/94294411-a6239500-ff14-11ea-8128-1559ae0bbe28.png)